### PR TITLE
Bug 1151803 - Serve the UI using gunicorn/WhiteNoise instead of Apache

### PR DESCRIPTION
**Pull request by edmorley**
In order that we can serve the UI on Heroku, we wrap the Django wsgi app with WhiteNoise, so both the UI and API requests are served by gunicorn.

In the Vagrant environment, Apache has been removed and Varnish instead now proxies all requests to gunicorn/Django runserver directly, without Apache as a go-between.

The UI on production will not be affected by this commit, since the Apache config there will still intercept requests for the UI assets rather than proxying them to gunicorn.

It's worth noting too, that we're not able to make use of WhiteNoise's automatic Django GZip/caching support since that assumes we are using Django templates and referring to resources using {% static "foo.css" %}

However, we can sub-class WhiteNoise (or more specifically the DjangoWhiteNoise class) and override the is_immutable_file() method to add caching support at a later date:
http://whitenoise.evans.io/en/latest/base.html#caching-headers

Documentation for WhiteNoise can be found at:
http://whitenoise.evans.io/

There are also a few related cleanup commits - view separately to keep the diffs cleaner :-)

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/639)

### CODE REVIEW

**Comment by maurodoglio** [comment]
This all looks good. The only thing I would try is to serve the ui files directly from WhiteNoise. The [whitenoise documentation](http://whitenoise.readthedocs.org/en/latest/base.html#using-whitenoise-with-any-wsgi-application) has an example on how to do that using WhiteNoise.add_files. It is probably worth a shot. I'm giving an r+ here in case you wanted to land this first.

**Comment by edmorley** [comment]
I've updated the PR to use add_files() - good suggestion - it actually means we get more control (and also saves us shoehorning the UI assets into the Django static folder on collectstatic.

The UI is now served at `/index.html` :-)

File: treeherder/webapp/wsgi.py [hunk 1]
```diff
@@ -18,7 +18,9 @@
 
 """
 import os
+from django.conf import settings
```
Comment (maurodoglio) [inline treeherder/webapp/wsgi.py]: I think you need to have DJANGO_SETTINGS_MODULE setup in your environment before you can safely import settings

Comment (edmorley) [inline treeherder/webapp/wsgi.py]: Yeah I was looking at that - I think we can drop that entirely, since it's only used for mod_wsgi with multiple sites.

**Comment by edmorley** [comment]
I've moved the definition of DJANGO_SETTINGS_MODULE higher up.

**Comment by maurodoglio** [comment]
:shipit:

**Pull request merged**